### ISSUE TITLE
Cluster got wonky for a few minutes on deployment of an application with large repository

### ISSUE THREAD

**Issue by philiplb** [flynn/flynn#2787]
Hi,

this ticket is maybe more a for your information.

I tried to deploy an application with a large git history. Large history means a git push of 1.11GB (there used to be some large binary files in the repo some years ago) which should result in a slug of the size ~155MB.

I pushed the application, the mvn build failed due to some missing dependencies. The last line of the git push is this:

> remote: Build failed: exec: job exited with status 1

And now the push hangs until I hit Ctrl+C. According to the status endpoint, this systems are unhealthy now in this moment:
- blobstore
- dashboard
- gitreceive
- logaggregator
- postgres

Everything recovered after a few minutes.

I tried to delete the application and got an error:

> timed out waiting for app deletion

Subsequent deletion tries lead to the same error.

Within this time, the other already deployed applications didn't answer or were utterly slow. Note: They don't use the DB resources Flynn offers but are talking to another DB server.

After a few more minutes, the application was gone as expected and everything started to work again.

Yesterday, I even broke the whole cluster by trying to deploy the application:
- Deletion threw a 503 and something else (I can't remember)
- I tried a reboot of a node
- basically all apps stopped to answer

I restored from backup and sadly don't have logs. Tried to reproduce this, but today, Flynn behaved fine.

**Comment by titanous** [comment]
This is due to the size of the repo, we're evaluating the best way to fix it.

**Comment by philiplb** [comment]
My current workaround, as I'm not interested in the history at all when deploying to Flynn:

```bash
mkdir deployment
rsync -avz --exclude '.git' --exclude 'deployment' --exclude 'target' . deployment
cd deployment
git init
git add .
git commit -m "deployment commit"
git remote add flynn https://git.flynn.xyz.com/myapp.git || true
git push flynn master
```

**Comment by lmars** [comment]
It would probably be helpful to support deploying tarballs #306, then you could do something like:

```
$ git archive | flynn push tarball
```

**Comment by lmars** [comment]
P.S. that is obviously not as performant, but probably a better workaround.

**Comment by tjboring** [comment]
I just experience a git push failure using the workaround mentioned by @philiplb 

Size of the repo using the workaround is 1866 objects.

**Comment by tjboring** [comment]
I've been able to solve this problem for my case by reducing the size of our git repo. I've removed some binaries that we're no longer using and some other static assets. This action helped reduce our slug size from 124M to 86M.

Since reducing the size of the repo, git push has been consistently successful, from my laptop and circleci. :)
